test/EndToEndTests/Services/ODataDefaultService/DefaultWCFSvcDataSource.cs

```diff
@@ -1205,15 +1206,15 @@ protected override IEdmModel CreateModel()
             return DefaultInMemoryModel.CreateODataServiceModel("Microsoft.Test.OData.Services.ODataWCFService");
         }
 
-        protected override void ConfigureContainer(IContainerBuilder builder)
+        protected override void ConfigureContainer(IServiceCollection services)
         {
-            base.ConfigureContainer(builder);
-            builder.AddServicePrototype(new ODataSimplifiedOptions()
+            base.ConfigureContainer(services);
+            services.AddSingleton(sp => new ServicePrototype<ODataSimplifiedOptions>(new ODataSimplifiedOptions()

```

Comment:
@robertmclaws  we have instances we have such: `            services.AddSingleton(sp => new ServicePrototype<ODataMessageReaderSettings>(new ODataMessageReaderSettings()));
            services.AddScoped(sp => sp.GetServicePrototype<ODataMessageReaderSettings>().Clone());`

Are you suggesting we do away with this? I believe the cloning is to make it easy to create copies of some of the services without affecting their current state. What will be a better alternative?

Reply:
Yep, I am definitely suggesting we remove this concept all the way up the stack. It's not consistent with DI implementations in other ASP.NET subsystems.

I'll branch this PR and work on a suggested option. 

I think the method we've adopted for Restier has served us well, where [we only add things](https://github.com/OData/RESTier/blob/main/src/Microsoft.Restier.AspNet.Shared/Extensions/RestierApiServiceCollectionExtensions.cs#L42-L102) if they haven't already been added.

Ideally, you would allow someone to configure the single instance that is created, instead of registering multiple instances of the same type and then having to clone, combine, or deal with FIFO/LIFO issues.

In situations where you have different settings for different routes, then you would use DI 8.0's new Keyed Services to replace the per-route container concept, and then register different settings instances for the route's key. DI resolution would look for the Keyed service first, then fall back to the default service.

This is why I pushed us to upgrade to DI 8.0 in earlier feedback... Keyed services solve a huge problem that will make OData on .NET 8 reduce memory usage, reduce allocations, and be crazy fast.

I'll work on it over the weekend and then we can PR review my suggestions separately :). You're awesome, have a great weekend 🤜🏻

Reply:
@robertmclaws  I think we can get rid of the `ServicePrototype` and register the instances directly but leave the cloning for now. We welcome a contribution on how to do away with the cloning.